src/spec_swagger/swagger2/core.cljc

```diff
@@ -1,6 +1,94 @@
 (ns spec-swagger.swagger2.core
   (:require [clojure.walk :as walk]
-            [spec-swagger.swagger2.schema :as schema]))
+            [clojure.spec.alpha :as s]
+            [spec-tools.json-schema :as json-schema]
+            [spec-tools.visitor :as visitor]
+            [spec-tools.core :as st]))
+
+;;
+;; conversion
+;;
+
+(defn- spec-dispatch [dispatch _ _ _] dispatch)
+(defmulti accept-spec spec-dispatch :default ::default)
+
+(defmethod accept-spec 'clojure.core/float? [_ _ _ _] {:type "number" :format "float"})
+(defmethod accept-spec 'clojure.core/double? [_ _ _ _] {:type "number" :format "double"})
+(defmethod accept-spec 'clojure.core/nil? [_ _ _ _] {})
+
+(defmethod accept-spec 'clojure.spec.alpha/or [_ _ children _]
+  ;; :anyOf is not supported by Swagger 2.0, so we just the the first child. In principle,
+  ;; we could do better in some special cases. For example, a reasonable schema
+  ;; for (s/or ::int int? ::str string?) would be {:type ["number", "string"]}.
+  (assoc
+    (first children)
```

Comment:
We have option to either:

1) use a convention to produce api-docs that can help to create valid requests. This fulfils the purpose: the first value is a valid value and an api-doc client & ui can be generated (but not all valid values can be sent via them). Using the first value is a convention from Schema & Ring-swagger, e.g. `(s/conditional s/Int pos?)` is an `s/Int` with some extra stuff on the tail.

2) we say "anything goes", but we produce invalid swagger spec (parameters need to have a type).

I would go with 1.